lib/Secrets/secrets_c.h

```diff
@@ -0,0 +1,150 @@
+    size_t dataSize;
+};
+
+struct Sailfish_Secrets_Secret_Identifier*
+Sailfish_Secrets_Secret_Identifier_new(
+        const char *name,
+        const char *collectionName);
+
+void Sailfish_Secrets_Secret_Identifier_delete(
+        struct Sailfish_Secrets_Secret_Identifier *ident);
+
+struct Sailfish_Secrets_Secret_FilterDatum*
+Sailfish_Secrets_Secret_FilterDatum_new(
+        const char *field,
+        const char *value);
+
+void Sailfish_Secrets_Secret_FilterDatum_delete(
+        struct Sailfish_Secrets_Secret_FilterDatum *filter);
+
+struct Sailfish_Secrets_Secret*
+Sailfish_Secrets_Secret_new(
+        const unsigned char *data,
+        size_t dataSize);
+
+void Sailfish_Secrets_Secret_delete(
+        struct Sailfish_Secrets_Secret *secret);
+
+void Sailfish_Secrets_Secret_setIdentifier(
+        struct Sailfish_Secrets_Secret *secret,
+        const char *name,
+        const char *collectionName);
+
+void Sailfish_Secrets_Secret_addFilter(
+        struct Sailfish_Secrets_Secret *secret,
+        const char *field,
+        const char *value);
+
+/****************************** secret manager **********************/
+
+enum Sailfish_Secrets_SecretManager_DeviceLockUnlockSemantic {
+    Sailfish_Secrets_SecretManager_DeviceLockKeepUnlocked = 0,
+    Sailfish_Secrets_SecretManager_DeviceLockRelock,
+};
+
+enum Sailfish_Secrets_SecretManager_AccessControlMode {
+    Sailfish_Secrets_SecretManager_OwnerOnlyMode = 0,
+    Sailfish_Secrets_SecretManager_SystemAccessControlMode
+};
+
+enum Sailfish_Secrets_SecretManager_UserInteractionMode {
+    Sailfish_Secrets_SecretManager_PreventInteraction = 0,
+    Sailfish_Secrets_SecretManager_SystemInteraction,
+    Sailflish_Secrets_SecretManager_ApplicationInteraction
+};
+
+int Sailfish_Secrets_SecretManager_createCollection(
+        const char *collectionName,
+        const char *storagePluginName,
+        const char *encryptionPluginName,
+        enum Sailfish_Secrets_SecretManager_DeviceLockUnlockSemantic unlockSemantic,
+        enum Sailfish_Secrets_SecretManager_AccessControlMode accessControlMode,
+        struct Sailfish_Secrets_Result **out_result);
+
+int Sailfish_Secrets_SecretManager_deleteCollection(
+        const char *collectionName,
+        enum Sailfish_Secrets_SecretManager_UserInteractionMode interactionMode,
+        struct Sailfish_Secrets_Result **out_result);
+
+int Sailfish_Secrets_SecretManager_setSecret(
+        struct Sailfish_Secrets_Secret *secret,
+        enum Sailfish_Secrets_SecretManager_UserInteractionMode uiMode,
+        const char *interactionServiceAddress,
+        struct Sailfish_Secrets_Result **out_result);
+
+int Sailfish_Secrets_SecretManager_getSecret(
+        struct Sailfish_Secrets_Secret_Identifier *ident,
+        enum Sailfish_Secrets_SecretManager_UserInteractionMode uiMode,
+        const char *interactionServiceAddress,
+        struct Sailfish_Secrets_Result **out_result,
+        struct Sailfish_Secrets_Secret **out_secret);
```

Comment:
Most operation (certainly get/set) can block, right? Then cancelable asynchronous version of each such call is needed too. I would even say required.